ScarletEditor/Sources/EditorMain.cpp

```diff
@@ -7,44 +7,341 @@
+				Lengths.push_back(scale);
+			}
+			if (x < width - 1 && y < height - 1)
+			{
+				Anchors.push_back(Grid[x + 1][y + 1]);
+				Lengths.push_back(sqrt2);
+			}
+
+			AddSpring(Grid[x][y], Anchors, Lengths, 100.f);
+		}
+	}
+
+	MakeGround();
+}
+
+void DemoSoftBodies(float heightOffset = 2.f)
+{
+	const int height = 3;
+	const int width = 3;
+	const int depth = 3;
+	float scale = 0.5f;
+	float offset = 2.f;
+	EID Grid[width][height][depth];
+	float sqrt2 = 1.414214f * scale;
+	float sqrt3 = 1.73205f * scale;
+
+
+	for (int x = 0; x < width; x++)
+	{
+		for (int y = 0; y < height; y++)
+		{
+			for (int z = 0; z < depth; z++)
+			{
+				Grid[x][y][z] = MakeSphereCollider(glm::vec3(x * scale - offset, y * scale + heightOffset, z * scale), glm::vec3(0.05f));
+			}
+		}
+	}
+
+	for (int x = 0; x < width; x++)
+	{
+		for (int y = 0; y < height; y++)
+		{
+			for (int z = 0; z < depth; z++)
+			{
+				std::vector<EID> Anchors;
+				std::vector<float> Lengths;
+
+				if (y < height - 1)
+				{
+					Anchors.push_back(Grid[x][y + 1][z]);
+					Lengths.push_back(scale);
+				}
+				if (x < width - 1)
+				{
+					Anchors.push_back(Grid[x + 1][y][z]);
+					Lengths.push_back(scale);
+				}
+				if (x < width - 1 && y < height - 1)
+				{
+					Anchors.push_back(Grid[x + 1][y + 1][z]);
+					Lengths.push_back(sqrt2);
+				}
+				if (z < depth - 1)
+				{
+					Anchors.push_back(Grid[x][y][z+1]);
+					Lengths.push_back(scale);
+				}
+				if (x < width - 1 && z < depth - 1)
+				{
+					Anchors.push_back(Grid[x + 1][y][z+1]);
+					Lengths.push_back(sqrt2);
+				}
+				if (x < width - 1 && y < height - 1 && z < depth - 1)
+				{
+					Anchors.push_back(Grid[x + 1][y + 1][z + 1]);
+					Lengths.push_back(sqrt3);
+				}
+
+				AddSpring(Grid[x][y][z], Anchors, Lengths, 1000.f);
+			}
+		}
+	}
+
+	MakeGround();
+}
+
+EID MakePlanet(glm::vec3 Pos, glm::vec3 V0, float Mass, float Scale, const char* Name)
+{
+	auto [Ent, Trans, Mesh, Rb] = GEditor->GetActiveWorld()->CreateEntity<Transform, StaticMeshComponent, RigidBodyComponent>(Name);
+
+	Trans->Position = Pos;
+	Trans->Rotation = glm::vec3(0.f, 0.f, 0.f);
+	Trans->Scale = glm::vec3(Scale * 0.05f);
+
+	Rb->Velocity = V0;
+	Rb->Mass = Mass;
+	Rb->UsesGravity = false;
+	Rb->UsesKeplerGravity = true;
+
+	MakeMesh(Mesh, "/ScarletEngine/Content/Sphere.obj");
+
+	return Ent->ID;
+};
+
+void DemoKepler()
+{
+	MakePlanet(glm::vec3(4.3f, -1.f, 0.f), glm::vec3(0.f, 0.f, 2.f), 1., 1.2f, "Jupiter");
+	MakePlanet(glm::vec3(3.f, 0.6f, 0.f), glm::vec3(0.f, 0.f, 2.5f), 1.f, 0.5f, "Venus");
+	MakePlanet(glm::vec3(4.15f, -0.4f, 0.f), glm::vec3(0., 0.f, 2.5f), 1.f, .3f, "Moon");
+	MakePlanet(glm::vec3(4.f, -0.4f, 0.f), glm::vec3(0.f, 0.f, 2.f), 7 * glm::pow(10, 8), 0.6f, "Earth");
+	MakePlanet(glm::vec3(2.f, 0.2f, 0.f), glm::vec3(0.f, 0.f, 2.5f), 1.f, 0.5f, "Mercury");
+	MakePlanet(glm::vec3(0.f, 0.f, 0.f), glm::vec3(0.f, 0.f, 0.f), 3 * glm::pow(10, 11), 2.f, "Sun");
+}
+
+#pragma endregion
+
```

Comment:
Instead of putting all of these in EditorMain.cpp, you could move them to a separate file in Achilles under the Achilles namespace (PhysicsDemos.h or smth) and if you want to run any of the demos you could include that and call the corresponding functions instead of cluttering this file.